transport/transport.go

```diff
@@ -215,6 +216,40 @@ type Stream struct {
 	statusDesc string
 }
 
+// Read reads bytes from the stream and manages flow control as needed.
+// This Read function blocks until either the len(p) bytes have been read
+// into p, or until an error occurs.
+// Note that err == nil iff n == len(p)
+// Also note that an error returned by this function indicates an error
+// in the underlying stream, and future calls to Read will continue to
+// return that same error.
+func (s *Stream) Read(p []byte) (n int, err error) {
+	if s.readFullErr != nil {
+		return 0, s.readFullErr
+	}
+	defer func() { s.readFullErr = err }()
+	s.streamWindowHandler(int64(len(p)))
+	return io.ReadFull(s.sr, p)
```

Comment:
Actually, FYI, return values are assigned via a return statement, so `return io.ReadFull(s.sr, p)` is fine.  Example: https://play.golang.org/p/V1qMks7o9s.